collectors/google_pubsub/pubsub_collector.go

```diff
@@ -36,19 +60,143 @@ func (collector *PubSubCollector) HandleMessage(message *pubsub.Message) error {
 	return result.Err
 }
 
+func (collector *PubSubCollector) PollForMessages() {
+	go collector.timeout()
+	for {
+		select {
+		case <-collector.ChannelToInitiatePollRequest:
+			if !collector.BaseCollector.Sleeping {
+				messages, err := collector.GetMessages()
+				if err != nil {
```

Comment:
Sorry just saw this - you should always call the `collector.ProcessExponentialBackoff` and send it the `err` if it's nil he can handle it and will process however it is needed by backing off or doing nothing.

Reply:
Updated!